### PR TITLE
kickoff v19 squid

### PR DESCRIPTION
**Pull request by rzarzynski**
The corresponding changeset in `ceph-build.git` is https://github.com/ceph/ceph-build/pull/2160.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/cephadm/cephadm.py
Comment (adk3798) [inline src/cephadm/cephadm.py:49]: just a note, the location of this constant is being moved as part of https://github.com/ceph/ceph/pull/53052. If that works get merged before this it might cause a conflict so wanted you to be aware of it. We plan to add a commit to that PR to change the release checklist to point to the new location once it's finalized as well.

Comment (rzarzynski) [reply src/cephadm/cephadm.py:49]: Thanks, Adam!

**Comment by rzarzynski** [comment]
CC: @yuvalif, @mattbenjamin.

File: src/include/ceph_features.h
Comment (idryomov) [reply src/include/ceph_features.h:255]: ... and the "include/ceph_features: retire deprecated bits" talks about nothing getting deprecated in mimic, which is pretty far from X-3 ;)

File: qa/workunits/test_telemetry_reef_x.sh
Comment (idryomov) [inline qa/workunits/test_telemetry_reef_x.sh:4]: This comment probably needs massaging.

cc @yaarith

File: qa/suites/upgrade/reef-x/filestore-remove-check/1-ceph-install/reef.yaml
Comment (idryomov) [inline qa/suites/upgrade/reef-x/filestore-remove-check/1-ceph-install/reef.yaml:10]: Does `filestore-remove-check` test make sense for `reef-x` suite?  Wouldn't this test be starting with a cluster that doesn't support filestore to begin with?

File: qa/suites/upgrade/reef-x/parallel/1-tasks.yaml
Comment (idryomov) [inline qa/suites/upgrade/reef-x/parallel/1-tasks.yaml:29]: Where would this workunit come from?  AFAICS this PR adds only `test_telemetry_reef_x.sh`.

File: qa/suites/upgrade/reef-x/parallel/1-tasks.yaml
Comment (idryomov) [inline qa/suites/upgrade/reef-x/parallel/1-tasks.yaml:3]: Nit: This commit's title says "bring qa/suites/upgrade/quincy-x".

File: qa/suites/upgrade/reef-x/parallel/1-tasks.yaml
Comment (rzarzynski) [reply qa/suites/upgrade/reef-x/parallel/1-tasks.yaml:29]: Good catch! There is also the need for `test_telemetry_reef.sh`. Adding.

File: qa/suites/upgrade/reef-x/filestore-remove-check/1-ceph-install/reef.yaml
Comment (rzarzynski) [reply qa/suites/upgrade/reef-x/filestore-remove-check/1-ceph-install/reef.yaml:10]: Yup, no need for it!

File: qa/suites/upgrade/reef-x/parallel/1-tasks.yaml
Comment (rzarzynski) [reply qa/suites/upgrade/reef-x/parallel/1-tasks.yaml:3]: Thanks! Fixed.

File: src/cephadm/cephadm.py
Comment (adk3798) [reply src/cephadm/cephadm.py:49]: this did end up getting merged. We've updated the checklist in the docs with the new location of this constant as part of that PR

**Comment by cbodley** [comment]
bump! a gentle reminder that https://github.com/ceph/ceph/pull/52114 is still blocked on the squid feature bit to add tracing information back to `MOSDOp`

**Comment by cbodley** [comment]
is just blocked on reviews? is it worth a round of testing in the meantime?

**Comment by cbodley** [comment]
can we consider splitting https://github.com/ceph/ceph/pull/53191/commits/b0a2db750bf7ade3c573015645a4a81421f446f6 into a separate PR to unblock https://github.com/ceph/ceph/pull/52114? @yuvalif does that require anything more than the feature bit?

**Comment by yuvalif** [comment]
> can we consider splitting [b0a2db7](https://github.com/ceph/ceph/commit/b0a2db750bf7ade3c573015645a4a81421f446f6) into a separate PR to unblock #52114? @yuvalif does that require anything more than the feature bit?

@cbodley, for the tracing changes I just need the feature bit

**Comment by cbodley** [comment]
> @cbodley, for the tracing changes I just need the feature bit

@yuvalif want to pull that single commit into your branch so we can make progress with testing? if it works there, we can try to get approval from @ceph/core to merge it

**Comment by rzarzynski** [comment]
@yuriw, @ljflores: is there a chance to give this PR some initial testing?

**Comment by yuriw** [comment]
@rzarzynski I can't build it so far

https://shaman.ceph.com/builds/ceph/wip-yuri10-testing-2023-11-22-1112/62cc3e177c7f02cb403e29ac74d80eb9203101ce/

**Comment by rzarzynski** [comment]
https://trello.com/c/NCZCeuVu/1894-wip-yuri10-testing-2023-11-22-1112

**Comment by rzarzynski** [comment]
Depends on https://github.com/ceph/ceph/pull/54721.

**Comment by rzarzynski** [comment]
Rebased just the sake of tagging (`v19.0.0`); no other changes.

**Review by adk3798 — APPROVED** [review]

**Review by cbodley — APPROVED** [review]

**Review by idryomov — APPROVED** [review]
Ack from the RBD perspective

**Review by vshankar — APPROVED** [review]
LGTM

**Comment by rzarzynski** [comment]
Related PRs in `ceph-container.git`:

* https://github.com/ceph/ceph-container/pull/2174,
* https://github.com/ceph/ceph-container/pull/2179 (need to be brought back when first Squid pkgs are out).

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcNCZCeuVu1894-wip-yuri10-testing-2023-12-12-1229-old-wip-yuri10-testing-2023-12-07-1728-old-wip-yuri10-testing-2023-12-05-1105-old-wip-yuri10

**Review by ljflores — APPROVED** [review]

**Pull request merged**
